src/content_mapblock.h

```diff
@@ -25,8 +25,25 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 struct MeshMakeData;
 struct MeshCollector;
 
-struct LightFrame
-{
+struct LightPair {
+	union {
+		u16 light;
+		struct {
+			u8 lightA;
+			u8 lightB;
+		};
+	};
+
+	LightPair() = default;
+	explicit LightPair(u16 value) : light(value) {}
```

Comment:
this is endian-dependent, probably not what you want
also the compiler might insert padding between `lightA` and `lightB` so the union thing is not even guaranteed to work

Reply:
> this is endian-dependent

Not a problem as long as you don’t mix bitwise operations with union member access. It is not guaranteed that `lightA` stores any specific bank, but it *is* guaranteed that it stores the same bank everywhere.

> the compiler might insert padding

Alignment requirements for a type can’t exceed its size, which is 1 byte in this case.

> the union thing is not even guaranteed to work

My mistake; according to cppreference.com such thing **is** guaranteed to work in C, but not in C++:
> If the member used to access the contents of a union is not the same as the member last used to store a value, the object representation of the value that was stored is reinterpreted as an object representation of the new type (this is known as type punning).

> it's undefined behavior to read from the member of the union that wasn't most recently written